prov/verbs/src/verbs_eq.c

```diff
@@ -861,6 +861,68 @@ vrb_eq_xrc_disconnect_event(struct vrb_eq *eq,
 	}
 }
 
+static inline int
+vrb_eq_addr_resolved_event(struct vrb_ep *ep)
+{
+	struct vrb_recv_wr *wr;
+	struct slist_entry *entry;
+	struct ibv_qp_init_attr attr = { 0 };
+	int ret;
+	struct vrb_cq *cq;
+	struct ibv_wc wc = {0};
+
+	assert(ofi_genlock_held(&vrb_ep2_progress(ep)->ep_lock));
+	assert(ep->state == VRB_RESOLVE_ADDR);
+
+	if (ep->util_ep.type == FI_EP_MSG) {
+		vrb_msg_ep_get_qp_attr(ep, &attr);
+
+		if (rdma_create_qp(ep->id, vrb_ep2_domain(ep)->pd, &attr)) {
+			ep->state = VRB_DISCONNECTED;
+			ret = -errno;
+			VRB_WARN(FI_LOG_EP_CTRL,
+				 "rdma_create_qp failed: %d\n", -ret);
+			return ret;
+		}
+
+		/* Allow shared XRC INI QP not controlled by RDMA CM
+		 * to share same post functions as RC QP. */
+		ep->ibv_qp = ep->id->qp;
+	}
+
+	assert(ep->ibv_qp);
+	while (!slist_empty(&ep->prepost_wr_list)) {
+		entry = slist_remove_head(&ep->prepost_wr_list);
+		wr = container_of(entry, struct vrb_recv_wr, entry);
+
+		ret = vrb_post_recv_internal(ep, &wr->wr);
+		if (ret && ep->util_ep.rx_cq) {
+			cq = container_of(ep->util_ep.rx_cq, struct vrb_cq, util_cq);
```

Comment:
I think on an error we can just return here.  Whatever path the failure above at line 885 takes should also flush any remaining preposted receives.